dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handler/DataHandler.java

```diff
@@ -232,39 +235,99 @@ void addIndicatorValues( DataQueryParams params, Grid grid )
 
             List<Indicator> indicators = resolveIndicatorExpressions( dataSourceParams );
 
-            // Try to get filters periods from dimension (pe), or else fall back
-            // to "startDate/endDate" periods
+            addIndicatorValues( params, dataSourceParams, indicators, grid );
+        }
+    }
+
+    /**
+     * Adds indicator values to the given grid based on the given data query
+     * parameters.
+     *
+     * @param params the {@link DataQueryParams}.
+     * @param grid the {@link Grid}.
+     */
+    void addExpressionDimensionItemValues( DataQueryParams params, Grid grid )
+    {
+        if ( !params.getExpressionDimensionItems().isEmpty() && !params.isSkipData() )
+        {
+            DataQueryParams dataSourceParams = newBuilder( params )
+                .retainDataDimension( EXPRESSION_DIMENSION_ITEM )
+                .withIncludeNumDen( false ).build();
+
+            List<ExpressionDimensionItem> expressionDimensionItems = resolveExpressionDimensionItemExpressions(
+                dataSourceParams );
+
+            List<Indicator> indicators = expressionDimensionItemsToIndicators( expressionDimensionItems );
+
+            addIndicatorValues( params, dataSourceParams, indicators, grid );
+        }
+    }
+
+    /**
+     * Transform expression dimension item object in the indicator object with
+     * some default values missing in expression dimension item
+     *
+     * @param expressionDimensionItems
+     * @return
+     */
+    private List<Indicator> expressionDimensionItemsToIndicators(
+        List<ExpressionDimensionItem> expressionDimensionItems )
+    {
+        return expressionDimensionItems.stream()
+            .map( edi -> {
+                IndicatorType indicatorType = new IndicatorType();
+                indicatorType.setNumber( true );
+                indicatorType.setFactor( 1 );
+
+                Indicator indicator = new Indicator();
+                indicator.setUid( edi.getUid() );
+                indicator.setName( edi.getName() );
+                indicator.setCode( edi.getCode() );
+                indicator.setIndicatorType( indicatorType );
+                indicator.setNumerator( edi.getExpression() );
+                indicator.setNumeratorDescription( edi.getDescription() );
+                indicator.setDenominator( "1" );
+                indicator.setDecimals( 0 );
```

Comment:
I think that we should bring the `decimals` to our meeting on Mondays and see how we should handle them.
Maybe users would like to choose the decimals, or maybe we should default to one or two decimals. Let's see...